scripts/generateAssetData/utils/portals.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  arbitrumChainId,
+  ASSET_NAMESPACE,
+  avalancheChainId,
+  baseChainId,
+  bscChainId,
+  type ChainId,
+  ethChainId,
+  gnosisChainId,
+  optimismChainId,
+  polygonChainId,
+  toAssetId,
+} from '@shapeshiftoss/caip'
+import type { Asset } from '@shapeshiftoss/types'
+import axios from 'axios'
+import qs from 'qs'
+import { getAddress, isAddressEqual, zeroAddress } from 'viem'
+
+import { colorMap } from '../colorMap'
+import { createThrottle } from '.'
+
+// Non-exhaustive - https://api.portals.fi/docs#/Supported/SupportedController_getSupportedTokensV2 for full docs
+type TokenInfo = {
+  key: string
+  name: string
+  decimals: number
+  symbol: string
+  address: string
+  images: string[] | undefined
+}
+
+type GetTokensResponse = {
+  totalItems: number
+  pageItems: number
+  more: boolean
+  page: number
+  tokens: TokenInfo[]
+}
+
+const CHAIN_ID_TO_PORTALS_NETWORK: Partial<Record<ChainId, string>> = {
+  [avalancheChainId]: 'avalanche',
+  [ethChainId]: 'ethereum',
+  [polygonChainId]: 'polygon',
+  [bscChainId]: 'bsc',
+  [optimismChainId]: 'optimism',
+  [arbitrumChainId]: 'arbitrum',
+  [gnosisChainId]: 'gnosis',
+  [baseChainId]: 'base',
+}
+
+export const fetchPortalsTokens = async (
+  chainId: ChainId,
+  page: number = 0,
+  accTokens: TokenInfo[] = [],
+): Promise<TokenInfo[]> => {
+  console.log(`Fetching page ${page} of Portals tokens for chainId: ${chainId}`)
```

Comment:
Revert me before opening PR, useful for debugging purposes

Reply:
Actually don't, this is pretty useful to get a grasp of something happening when devving locally, especially with the high amount of requests